### PR TITLE
Improve docgo generation in Makefile

### LINKED ISSUE #147: Improve docgo generation
As developer, is very annoying to see the `go.mod` and `go.sum` modified every time that the `docgo` generated files are updated.

### Acceptance criteria

- [ ] The `docgo` execution doesn't modify `go.sum` or `go.mod` files

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Minor update on Makefile to avoid modification of `go.mod` and `go.sum` when running `docgo`

Fixes #147 

## Type of change

- Refactor (refactoring code, removing useless files)
- Documentation update

## Testing steps

Regular CI

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,9 +67,8 @@ test: clean build ## Run the unit tests
 cover: test
 	@go tool cover -html=coverage.out
 
-license:
-	GO111MODULE=off go get -u github.com/google/addlicense && \
-		addlicense -c "Red Hat, Inc" -l "apache" -v ./
+license: install_addlicense
+	addlicense -c "Red Hat, Inc" -l "apache" -v ./
```
Comment (Sergey1011010) [inline Makefile:71]: Why did you remove GO111MODULE=off?

File: Makefile [hunk 2]
```diff
@@ -83,8 +85,15 @@ help: ## Show this help screen
...
+godoc: export GO111MODULE=off
+godoc: install_docgo install_addlicense ${DOCFILES}
 
-openapi-check:
-	./check_openapi.sh
+install_docgo: export GO111MODULE=off
+install_docgo:
+	[[ `command -v docgo` ]] || GO111MODULE=off go get -u github.com/dhconnelly/docgo
+
+install_addlicense: export GO111MODULE=off
```
Comment (Sergey1011010) [inline Makefile:97]: It's will be run before addlicense by makefile's magic? Did I get it correctly?

Comment (joselsegura) [reply Makefile:97]: Makefile's magic will export that environment variable to the execution of the rule `install_addlicense`, so the execution of that rule will be covered by this env var definition.

It answers your previous comment. I remove `GO111MODULE` from one side and put in a different place. It has more sense with `godoc` rule, that will avoid command `docgo` to update `go.sum` and `go.mod`.

Comment (Sergey1011010) [reply Makefile:97]: Thanks
